### DISCUSSION TITLE
run echo hello > foo.txt - File is empty

### DISCUSSION THREAD

**Discussion by velle** [bats-core/bats-core#950] [Q&A]
Bats file:

    @test "dummy 1" {
      run echo hello > foo.txt
    }

Shell lines:

    $ echo hola > foo.txt
    $ bats hello.bats
     ✓ dummy 1

    1 test, 0 failures
    $ cat foo.txt
    $ 

If foo.txt does not exist, it is created. It's as if the stdout from echo is empty. I also tried with cat, again the stdout seems to be empty. 

What is going on? :) I'm sure there is a good explanation. 

How can I write a text string to a file from within a bats file?

Sincerely

**Comment by martin-schulze-vireso** [comment]
Similar to https://bats-core.readthedocs.io/en/stable/gotchas.html#my-piped-command-does-not-work-under-run because of parsing rules, the redirect applies to `run`, not the `echo`.

You could put that into its own function.

However, in this simple case I would just omit `run`.
